### PR TITLE
Remove default postgres driver and use PGX everywhere

### PR DESCRIPTION
**Pull request by alishakawaguchi**

### CODE REVIEW

File: backend/pkg/sqlmanager/postgres/postgres-manager.go [hunk 1]
```diff
@@ -15,6 +15,8 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+const PostgresDriver = "postgres"
```
Comment (nickzelei) [inline backend/pkg/sqlmanager/postgres/postgres-manager.go:18]: does this need to be exposed outside of this module?

File: backend/pkg/dbconnect-config/postgres.go [hunk 2]
```diff
@@ -8,9 +8,10 @@ import (
 
 	mgmtv1alpha1 "github.com/nucleuscloud/neosync/backend/gen/go/protos/mgmt/v1alpha1"
 	"github.com/nucleuscloud/neosync/backend/pkg/clienttls"
-	sqlmanager_shared "github.com/nucleuscloud/neosync/backend/pkg/sqlmanager/shared"
 )
 
+const PostgresDriver = "postgres"
```
Comment (nickzelei) [inline backend/pkg/dbconnect-config/postgres.go:13]: does this need to be exposed outside of this module?

File: backend/pkg/sqlmanager/postgres/postgres-manager.go (hunk 1 above)
Comment (nickzelei) [reply backend/pkg/sqlmanager/postgres/postgres-manager.go:18]: In the querybuilder we have a `GoquPostgresDriver`, should be we be using that on 878?

Comment (alishakawaguchi) [reply backend/pkg/sqlmanager/postgres/postgres-manager.go]: Import cycle 😢 . moved it to the shared sqlmanager. So now we just have PostgresDriver and GoquPostgresDriver

File: backend/pkg/sqlmanager/shared/types.go [hunk 3]
```diff
@@ -99,10 +99,9 @@ func ToConstraintType(constraintType string) (ConstraintType, error) {
 }
 
 const (
-	PostgresDriver        = "pgx"
-	DefaultPostgresDriver = "postgres"
-	MysqlDriver           = "mysql"
-	MssqlDriver           = "sqlserver"
+	PostgresDriver = "pgx"
```
[the remark this answers is not in the document]
Comment (alishakawaguchi) [reply backend/pkg/sqlmanager/shared/types.go]: I removed all the other postgresdrivers

**Review by nickzelei — APPROVED** [review]

**Pull request merged**